chain-storage-sqlite-old/src/lib.rs

```diff
@@ -3,16 +3,31 @@ use chain_storage::{
     error::Error,
     store::{BackLink, BlockInfo, BlockStore},
 };
-use r2d2_sqlite::SqliteConnectionManager;
-use rusqlite::{types::Value, OpenFlags};
-use std::path::Path;
+use rusqlite::{types::Value, Connection, OpenFlags};
+use std::path::{Path, PathBuf};
+
+enum StoreType {
+    Memory,
+    File(PathBuf),
+}
 
-#[derive(Clone)]
 pub struct SQLiteBlockStore<B>
```

Comment:
I wonder if the generic parameter can be removed from this type, as only the connection and the `connect` method really require it. This could necessitate turbofishing on `connect`, but otherwise would declutter the code.